sqlite.go

```diff
@@ -540,7 +548,13 @@ func (s *stmt) QueryContext(ctx context.Context, args []driver.NamedValue) (driv
 	if err := s.bindAll(args); err != nil {
 		return nil, err
 	}
-	return &rows{stmt: s}, nil
+	cancel := func() {}
+	if ctx.Value(queryCancelKey{}) != nil {
+		ctx, cancel = context.WithCancel(ctx)
+		db := s.stmt.DBHandle()
+		go func() { <-ctx.Done(); db.Interrupt() }()
```

Comment:
similarly here I think if you define `*rows` before the if statement, you could condition calling Interrupt on `rows.closed`

Reply:
I think if I do that I'll have to add a lock or an atomic on closed, which would otherwise become racy.

Reply:
oof yeah you're right.

taking a quick read of the sqlite3 sources, sqlite3_interrupt is an atomic store, that is unconditionally stored over when _step is called. so 99% of the cost of unconditionally calling interrupt is just the cgo overhead. given this path adds a goroutine, it's not worth optimizing. so this is fine as is.